next branch issues running on Ubuntu 20.04

The +experiment=simple_dqn/cartpole crashes with the following error on Ubuntu:

[2022-08-29 12:55:38,295][cogment_verse.app][INFO] - Run starting...
[2022-08-29 12:55:38,328][cogment_verse.processes.run][INFO] - Starting run [pedantic_hofstadter_0] from [actors.simple_dqn.SimpleDQNTraining]
[2022-08-29 12:55:38,727][cogment_verse.mlflow_experiment_tracker][INFO] - Experiment with name '/actors.simple_dqn.SimpleDQNTraining' not found. Creating it.
[2022-08-29 12:55:41,056][cogment_verse.processes.run][ERROR] - Error while executing run [pedantic_hofstadter_0] from [actors.simple_dqn.SimpleDQNTraining]
Traceback (most recent call last):
  File "/home/vahid/Desktop/cogment-verse/dev_dagger/cogment_verse/processes/run.py", line 64, in run_main_async
    await run.impl(run_session)
  File "/home/vahid/Desktop/cogment-verse/dev_dagger/actors/simple_dqn.py", line 367, in impl
    run_session.log_metrics(trial_idx, total_reward=total_reward)
  File "/home/vahid/Desktop/cogment-verse/dev_dagger/cogment_verse/run/run_session.py", line 84, in log_metrics
    self._xp_tracker.log_metrics(step_timestamp=int(time.time() * 1000), step_idx=self._step_idx, *args, **kwargs)
TypeError: log_metrics() got multiple values for argument 'step_timestamp'

I have included the changes from pytorch_multiproc_fix branch.

I also see inconsistent behaviour with the +experiment=simple_a2c/cartpole. It sometimes runs fine, and sometimes stalls after running a bunch of epochs with no error.

--- a/actors/simple_dqn.py
+++ b/actors/simple_dqn.py
@@ -364,7 +364,7 @@ class SimpleDQNTraining:
             trial_done = done.item() == 1
 
             if trial_done:
-                run_session.log_metrics(trial_idx, total_reward=total_reward)
+                run_session.log_metrics(trial_idx=trial_idx, total_reward=total_reward)
                 total_reward_cum += total_reward
                 if (trial_idx + 1) % 100 == 0:
                     total_reward_avg = total_reward_cum / 100

--- a/cogment_verse/mlflow_experiment_tracker.py
+++ b/cogment_verse/mlflow_experiment_tracker.py
@@ -133,10 +133,10 @@ class MlflowExperimentTracker:
             params=[Param(key, str(value)) for key, value in make_dict(False, *args, **kwargs).items()],
         )
 
-    def log_metrics(self, step_timestamp, step_idx, *args, **kwargs):
+    def log_metrics(self, step_timestamp, step_idx, **kwargs):
         EXPERIMENT_TRACKER_METRICS_LOGGED_COUNTER.inc(len(kwargs))
         self._metrics_buffer.extend(
-            [Metric(key, value, step_timestamp, step_idx) for key, value in make_dict(True, *args, **kwargs).items()]
+            [Metric(key, value, step_timestamp, step_idx) for key, value in make_dict(True, **kwargs).items()]
         )
         self._start_flush_metrics_worker()
 

--- a/cogment_verse/run/run_session.py
+++ b/cogment_verse/run/run_session.py
@@ -80,8 +80,8 @@ class RunSession:
     def log_params(self, *args, **kwargs):
         self._xp_tracker.log_params(*args, **kwargs)
 
-    def log_metrics(self, *args, **kwargs):
-        self._xp_tracker.log_metrics(step_timestamp=int(time.time() * 1000), step_idx=self._step_idx, *args, **kwargs)
+    def log_metrics(self, **kwargs):
+        self._xp_tracker.log_metrics(step_timestamp=int(time.time() * 1000), step_idx=self._step_idx, **kwargs)
 
     def terminate_failure(self):
         self._xp_tracker.terminate_failure()

